addons/account_edi/models/account_edi_format.py

```diff
@@ -324,6 +324,10 @@ def _create_document_from_attachment(self, attachment):
                         edi_format.name,
                         str(e))
                 if res:
+                    for move in res:
+                        if move.move_type in ['in_invoice', 'in_refund']:
+                            references = [move.invoice_origin] if move.invoice_origin else []
+                            move._find_and_set_purchase_orders(references, move.partner_id.id, move.amount_total, timeout=4)
```

Comment:
Maybe in a separated method because: duplicated code AND I think logic could be extended, for example in Italy there might be more than one reference.